src/Transition.js

```diff
@@ -303,7 +438,7 @@ class Transition extends React.Component {
     }
 
     const {children, ...childProps} = this.props;
-    Object.keys(Transition.propTypes).forEach(key => delete childProps[key]);
+    Object.keys(propTypes).forEach(key => delete childProps[key]);
```

Comment:
lets do `delete props.foo` for each one manually instead, its a bit uglier but maintains the optimization